sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/GsonJsonArray.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.gson;
+
+import com.azure.core.util.serializer.JsonArray;
+import com.azure.core.util.serializer.JsonNode;
+import com.google.gson.JsonElement;
+
+import java.util.Objects;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * GSON specific implementation of {@link JsonArray}.
+ */
+public final class GsonJsonArray implements JsonArray {
+    private final com.google.gson.JsonArray jsonArray;
+
+    /**
+     * Constructs a {@link JsonArray} backed by an empty GSON {@link com.google.gson.JsonArray}.
+     */
+    public GsonJsonArray() {
+        this.jsonArray = new com.google.gson.JsonArray();
+    }
+
+    /**
+     * Constructs a {@link JsonArray} backed by the passed GSON {@link com.google.gson.JsonArray}.
+     *
+     * @param jsonArray The backing GSON {@link com.google.gson.JsonArray}.
+     */
+    public GsonJsonArray(com.google.gson.JsonArray jsonArray) {
+        this.jsonArray = jsonArray;
```

Comment:
Should we do a `null` check on all ctrs of all core.json type that takes gson types (same for jackson)? later methods in these types assume the instance it wraps is not null

Reply:
Will add `Objects.requireNonNull` checks to the constructors.